crates/types/src/invocation.rs

```diff
@@ -159,6 +162,15 @@ pub enum ServiceInvocationResponseSink {
     Ingress(IngressDispatcherId),
 }
 
+/// Caller of an invocation
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum Caller {
+    Ingress,
+    Service(FullInvocationId),
+    /// Internal calls for the non-deterministic built-in services
+    Internal,
```

Comment:
Is there any metadata that we can add about the "Internal" case?

Reply:
Anything interesting I can currently think of should be contained in the `ServiceInvocation` already (which method is being called, timestamps). Maybe we'll add more information as we'll discover the need for it.